tls/s2n_resume.c

```diff
@@ -417,7 +421,8 @@ int s2n_resume_from_cache(struct s2n_connection *conn)
     if (result == S2N_CALLBACK_BLOCKED) {
         POSIX_BAIL(S2N_ERR_ASYNC_BLOCKED);
     }
-    POSIX_GUARD(result);
+
+    POSIX_ENSURE(result >= S2N_SUCCESS, S2N_ERR_CANCELLED);
```

Comment:
Nit: newline here definitely isn't good for readability. Let's handle result in the same block where it was set.